### PR TITLE
feat: Add proves verification in integration tests

### LINKED ISSUE #25: Fix inconsistencies between nmt-rs and go-nmt proofs

### PR DESCRIPTION
**Pull request by zvolin**
closes #25

### CODE REVIEW

File: rpc/tests/blob.rs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 use std::time::Duration;
 
 use celestia_rpc::prelude::*;
-use celestia_types::{Blob, Commitment};
+use celestia_types::{nmt::NamespacedHash, Blob, Commitment};
```
Comment (oblique) [inline rpc/tests/blob.rs:4]:
```suggestion
use celestia_types::nmt::NamespacedHash;
use celestia_types::{Blob, Commitment};
```

File: types/src/blob.rs [hunk 2]
```diff
@@ -34,15 +36,18 @@ impl Blob {
     }
 
     pub fn validate(&self) -> Result<()> {
-        let computed_commitment =
-            Commitment::generate(self.namespace, self.share_version, &self.data)?;
+        let computed_commitment = Commitment::generate(self.namespace, self.to_shares()?)?;
```
Comment (oblique) [inline types/src/blob.rs:39]: I think I prefer the following API: 

* Rename old `Commitment::generate` to `Commitment::with_blob_data` (or `Commitment::from_blob_data`?)
* Add `Commitment::with_shares` (or `Commitment::from_shares`?)

Comment (zvolin) [reply types/src/blob.rs:39]: I can just leave the `Commitment::generate` if you prefer, we don't need the version from shares

Comment (oblique) [reply types/src/blob.rs:39]: Yes, I prefer the older API

Comment (zvolin) [reply types/src/blob.rs]: reverted

**Comment by oblique** [comment]
Closing because we incorporated the changes in #24

**Pull request closed without merging**